### PR TITLE
Feat: Authenticate user with credentials

### PR DESCRIPTION
**Pull request by nihal-dhore**
- Integrated custom signin and signup forms with error handling with next-auth.
- Implemented zod for credentials validation. 
- Changed file structure a bit.
- Implemented prisma singleton.

image

image

### CODE REVIEW

File: next-app/app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -1,6 +1,163 @@
...
+import { JWT } from "next-auth/jwt";
+import bcrypt from "bcryptjs";
 
-const handler = NextAuth(authOptions);
+import prismaClient from "@/lib/db";
+import { PrismaClientInitializationError } from "@prisma/client/runtime/library";
+import { emailSchema, passwordSchema } from "@/schema/credentials-schema";
 
-export { handler as GET, handler as POST };
+const handler = NextAuth({
```
Comment (SujithThirumalaisamy) [inline next-app/app/api/auth/[...nextauth]/route.ts:11]: Can you move this to the `@/lib/auth-options` file again. Fix the type issues by using `satisfies NextAuthOptions`

File: next-app/app/api/auth/[...nextauth]/route.ts [hunk 2]
```diff
@@ -1,6 +1,163 @@
...
+        }
+
+        try {
+          const user = await prismaClient.user.findUnique({
+            where: {
+              email: emailValidation.data
+            }
+          });
+
+          //console.log(user);
```
Comment (SujithThirumalaisamy) [inline next-app/app/api/auth/[...nextauth]/route.ts:46]: Always remove logs

File: next-app/app/api/auth/[...nextauth]/route.ts (hunk 1 above)
Comment (SujithThirumalaisamy) [reply next-app/app/api/auth/[...nextauth]/route.ts:11]: You have the authoptions in the file. Just use that

File: next-app/app/api/auth/[...nextauth]/route.ts (hunk 2 above)
Comment (nihal-dhore) [reply next-app/app/api/auth/[...nextauth]/route.ts]: done

File: next-app/app/api/auth/[...nextauth]/route.ts (hunk 1 above)
Comment (nihal-dhore) [reply next-app/app/api/auth/[...nextauth]/route.ts]: done

**Pull request merged**